### PR TITLE
Feature/game basics

### PR DESCRIPTION
**Pull request by Jharaxus**
Add basic game components and data structures

### CODE REVIEW

**Review by Kaz-ookid — COMMENTED** [review]
Very nice features! The implementation is basic and it offers a comfortable base for a lot of new feature to use. 

It looks good to me as it is, given that it is necessary roots for further features. In my opinion a merge is not yet necessary, as branches that need it can just get it here for the moment. The choices of implementation are important to be well thought and discussed before a merge.

**Review by Kaz-ookid — COMMENTED** [review]
You should consider adding `@throws` above the functions that throw exceptions. This is helpfull, as it forces us to catch them when using your functions :)

File: app/src/main/java/com/github/polypoly/app/game/PendingGame.kt
Comment (Jharaxus) [reply app/src/main/java/com/github/polypoly/app/game/PendingGame.kt:21]: Yes you are right I will change that

File: app/src/main/java/com/github/polypoly/app/game/PendingGame.kt
Comment (Jharaxus) [reply app/src/main/java/com/github/polypoly/app/game/PendingGame.kt:37]: I hesitated on that one but I thought that as the number of players is publicly available, it should be the caller's responsability (and more specifically probably in our case the joining activity) to ensure that no user attemps to join a game already full. This is the reason why I find it more logic to response to such incorrect behavior with an exception rather than a boolean.
It might be more efficient and less verbose to use a boolean instead when this will actually be used, however I still believe it makes more sense as it is for now :)

**Comment by Jharaxus** [comment]
> Very nice features! The implementation is basic and it offers a comfortable base for a lot of new feature to use.
> 
> It looks good to me as it is, given that it is necessary roots for further features. In my opinion a merge is not yet necessary, as branches that need it can just get it here for the moment. The choices of implementation are important to be well thought and discussed before a merge.

I get your point but then if anybody who needs it merges that branch then it's kind of like this branch was the new main, and it would prevent any the branches using that one from being able to merge with main.
Anyway if we need do to some modifications to this code (which will most likely happen), we could just create another PR for it, but in the meantime we are not blocked by this PR.

**Comment by Kaz-ookid** [comment]
LGTM!

**Review by Kaz-ookid — APPROVED** [review]

**Pull request merged**
